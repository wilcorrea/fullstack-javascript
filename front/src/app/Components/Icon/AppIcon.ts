import { CreateElement, RenderContext } from 'vue'

/**
 * @type {string}
 */
const block = 'app-icon'

/**
 * @component {AppIcon}
 */
export default {
  /**
   */
  functional: true,
  /**
   */
  name: 'AppIcon',
  /**
   * @param {CreateElement} h
   * @param {RenderContext} context
   */
  render (h: CreateElement, context: RenderContext) {
    const { props, listeners } = context
    const style = {}
    if (props.color) {
      // @ts-ignore
      style['color'] = props.color
    }
    if (listeners.click) {
      // @ts-ignore
      style['cursor'] = 'pointer'
    }
    const data = { class: `${block} material-icons`, style, on: { ...listeners } }
    let name = props.name
    if (!name) {
      name = 'warning'
      console.warn('[app-icon]', ' ', 'using a fallback icon, please provide a name')
    }

    return h('span', data, name)
  }
}
