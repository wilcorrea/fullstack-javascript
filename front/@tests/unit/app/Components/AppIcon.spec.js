import { shallowMount } from '@vue/test-utils'
import AppIcon from 'src/app/Components/Icon/AppIcon'

describe('AppIcon', () => {
  const name = 'vue'

  /**
   * @test
   */
  it('try without a name', () => {
    const wrapper = shallowMount(AppIcon)
    expect(wrapper.html()).toMatch(
      '<span class="app-icon material-icons">warning</span>'
    )
  })

  /**
   * @test
   */
  it('check if click produce cursor pointer', () => {
    const wrapper = shallowMount(
      AppIcon, { propsData: { name }, listeners: { click: () => undefined } }
    )
    expect(wrapper.html()).toMatch(
      `<span class="app-icon material-icons" style="cursor: pointer;">${name}</span>`
    )
  })

  const color = 'black'
  const wrapper = shallowMount(AppIcon, { propsData: { name, color } })
  /**
   * @test
   */
  it('create a simple icon', () => {
    expect(wrapper.html()).toMatch(
      `<span class="app-icon material-icons" style="color: ${color};">${name}</span>`
    )
  })

  /**
   * @test
   */
  test('should render content correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
