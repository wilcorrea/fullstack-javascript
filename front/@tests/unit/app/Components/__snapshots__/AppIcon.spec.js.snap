// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppIcon should render content correctly 1`] = `<span class="app-icon material-icons" style="color: black;">vue</span>`;
